cvat-ui/src/components/quality-control/quality-settings-tab.tsx

```diff
@@ -2,41 +2,93 @@
+    instance: Task | Project;
     fetching: boolean;
-    qualitySettings: QualitySettings | null;
-    setQualitySettings: (settings: QualitySettings) => void;
+    qualitySettings: {
+        settings: QualitySettings | null;
+        childrenSettings: QualitySettings[] | null;
+    };
+    setQualitySettings: (settingsList: QualitySettings[], onError?: () => void) => void;
 }
 
 function QualitySettingsTab(props: Readonly<Props>): JSX.Element | null {
     const {
+        instance,
         fetching,
-        qualitySettings: settings,
+        qualitySettings: { settings, childrenSettings },
         setQualitySettings,
     } = props;
 
     const [form] = Form.useForm();
+
+    const setupSettingsFromForm = useCallback(async (newSettings: QualitySettings) => {
+        const values = await form.validateFields();
+
+        newSettings.targetMetric = values.targetMetric;
+        newSettings.targetMetricThreshold = values.targetMetricThreshold / 100;
+
+        newSettings.maxValidationsPerJob = values.maxValidationsPerJob;
+
+        newSettings.lowOverlapThreshold = values.lowOverlapThreshold / 100;
+        newSettings.iouThreshold = values.iouThreshold / 100;
+        newSettings.compareAttributes = values.compareAttributes;
+        newSettings.emptyIsAnnotated = values.emptyIsAnnotated;
+
+        newSettings.oksSigma = values.oksSigma / 100;
+        newSettings.pointSizeBase = values.pointSizeBase;
+
+        newSettings.lineThickness = values.lineThickness / 100;
+        newSettings.lineOrientationThreshold = values.lineOrientationThreshold / 100;
+        newSettings.orientedLines = values.orientedLines;
+
+        newSettings.compareGroups = values.compareGroups;
+        newSettings.groupMatchThreshold = values.groupMatchThreshold / 100;
+
+        newSettings.checkCoveredAnnotations = values.checkCoveredAnnotations;
+        newSettings.objectVisibilityThreshold = values.objectVisibilityThreshold / 100;
+
+        newSettings.panopticComparison = values.panopticComparison;
+        newSettings.jobFilter = values.jobFilter || '';
+    }, [form]);
+
     const onSave = useCallback(async () => {
-        try {
-            const values = await form.validateFields();
-            setQualitySettings(values);
-        } catch (error) {
-            notification.error({
-                message: 'Could not save quality settings',
-                description: formFieldsError(error).map((text: string): JSX.Element => <div>{text}</div>),
-                className: 'cvat-notification-save-quality-settings-failed',
+        if (settings) {
+            await setupSettingsFromForm(settings);
+            setQualitySettings([settings]);
+        }
+    }, [form, settings, setQualitySettings]);
+
+    const onInheritChange = useCallback(async (value: boolean) => {
+        if (settings) {
+            settings.inherit = value;
+            setQualitySettings([settings], () => {
+                settings.inherit = !value;
```

Comment:
Avoiding such handlers is one of reasons why I suggested to rework qualitySettings.save method

Reply:
Removed